core/new-gui/src/app/workspace/component/workflow-editor/mini-map/mini-map.component.ts

```diff
@@ -1,247 +1,75 @@
-   */
-  private handleWindowResize(): void {
     fromEvent(window, "resize")
       .pipe(auditTime(30))
       .pipe(untilDestroyed(this))
-      .subscribe(() => {
-        this.updateMiniMapScaleSize();
-        this.updateNavigatorOffset();
-        this.updateNavigatorDimension();
-      });
-  }
-
-  private updateNavigatorOffset(): void {
-    // set navigator position in the component
-    const mainPaperPoint = this.workflowActionService
-      .getJointGraphWrapper()
-      .pageToJointLocalCoordinate(this.getMainPaperWrapperElementOffset());
-    const miniMapPoint = this.mainPaperToMiniMapPoint(mainPaperPoint);
-    const miniMapWrapperOffset = jQuery("#" + this.MINI_MAP_WRAPPER_ID).offset();
-    if (!miniMapWrapperOffset) {
-      throw new Error("cannot find minimap wrapper");
-    }
-    const left = miniMapWrapperOffset.left + miniMapPoint.x;
-    const top = miniMapWrapperOffset.top + miniMapPoint.y;
-    jQuery("#" + this.MINI_MAP_NAVIGATOR_ID).css({
-      left: left + "px",
-      top: top + "px",
-    });
-  }
-
-  /**
-   * This method sets the dimension of the navigator based on the browser size.
-   */
-  private updateNavigatorDimension(): void {
-    const { width: mainPaperWidth, height: mainPaperHeight } = this.getOriginalWrapperElementSize();
-    const mainPaperScale = this.workflowActionService.getJointGraphWrapper().getMainJointPaper()?.scale() ?? {
-      sx: 1,
-      sy: 1,
-    };
-
-    // set navigator dimension size, mainPaperDimension * MINI_MAP_ZOOM_SCALE is the
-    //  main paper's size in the mini-map
-    const width = (mainPaperWidth / mainPaperScale.sx) * this.MINI_MAP_ZOOM_SCALE;
-    const height = (mainPaperHeight / mainPaperScale.sy) * this.MINI_MAP_ZOOM_SCALE;
-    jQuery("#" + this.MINI_MAP_NAVIGATOR_ID).width(width);
-    jQuery("#" + this.MINI_MAP_NAVIGATOR_ID).height(height);
+      .subscribe(() => this.updateNavigator());
   }
 
-  private getMainPaperWrapperElementOffset(): Point {
-    const offset = jQuery("#" + this.MAIN_PAPER_JOINTJS_WRAPPER_ID).offset();
-    if (offset === undefined) {
-      throw new Error("fail to get Workflow Editor wrapper element offset");
-    }
-    return { x: offset.left, y: offset.top };
-  }
-
-  /**
-   * This method gets the original paper wrapper size.
-   */
-  private getOriginalWrapperElementSize(): { width: number; height: number } {
-    let width = jQuery("#" + this.MAIN_PAPER_JOINTJS_WRAPPER_ID).width();
-    let height = jQuery("#" + this.MAIN_PAPER_JOINTJS_WRAPPER_ID).height();
-
-    // when testing, width and height will be undefined, this gives default value
-    //  according to css grids
-    width = width === undefined ? window.innerWidth * 0.7 : width;
-    height = height === undefined ? window.innerHeight - 56 - 25 : height;
-
-    return { width, height };
+  private updateNavigator(): void {
+    const mainPaper = this.workflowActionService.getJointGraphWrapper().getMainJointPaper()!;
```

Comment:
try to avoid using `!` if possible. if needed, we can make `.getMainJointPaper()` return non-undefine?

Reply:
Fixed.